plugins/janus_videoroom.c

```diff
@@ -6884,9 +6884,16 @@ static void *janus_videoroom_handler(void *data) {
 			} else if(!strcasecmp(request_text, "start")) {
 				/* Start/restart receiving the publisher streams */
 				if(subscriber->paused && msg->jsep == NULL) {
+					janus_videoroom_publisher *feed = subscriber->feed;
+
 					/* This is just resuming a paused stream, reset the RTP sequence numbers */
 					subscriber->context.a_seq_reset = TRUE;
 					subscriber->context.v_seq_reset = TRUE;
+				
+					if(feed && feed->sdp && strstr(feed->sdp, "m=video")) {
+						/* Send a FIR */
+						janus_videoroom_reqpli(feed, "Subscriber start");
+					}
```

Comment:
I don't think that works, if the purpose is only ensuring quicker recovery after a resume. It should be something like this instead:

```
if(subscriber->paused && feed) {
	/* Send a FIR */
	janus_videoroom_reqpli(feed, "Subscriber resumed");
}
```

The way the patch is conceived right now, it would _only_ work when first subscribing, which is exactly when the PLI should _not_ be sent (the PeerConnection doesn't exist yet).

Reply:
Ah, scratch that, I'm an idiot, I was looking at the wrong thing...

Reply:
That said, I think the if check can be simplified in

```
if(feed && g_atomic_int_get(&feed->started, 1)) {
```

as all we care about is if the PC is up. If there's no video, the PLI will be ignored by the core.

Reply:
done

Reply:
Ah sorry, the atomic check is a typo, my fault, it should be

```
g_atomic_int_get(&feed->started)
```

The extra argument is only needed for set.

Reply:
fixed